statsdreceiver: Test_statsdreceiver_EndToEnd and Test_Server_ListenAndServe are unstable

See https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-collector-contrib/5686/workflows/147bbc6a-664d-4402-b869-53eb0fba3ede/jobs/43784

```
--- FAIL: Test_statsdreceiver_EndToEnd (0.00s)
    --- FAIL: Test_statsdreceiver_EndToEnd/default_config (0.00s)
        receiver_test.go:133: 
            	Error Trace:	receiver_test.go:133
            	Error:      	Received unexpected error:
            	            	listen udp 127.0.0.1:50409: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
            	Test:       	Test_statsdreceiver_EndToEnd/default_config
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver	0.048s
ok  	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/protocol	0.036s
--- FAIL: Test_Server_ListenAndServe (0.01s)
    --- FAIL: Test_Server_ListenAndServe/udp (0.01s)
        server_test.go:54: 
            	Error Trace:	server_test.go:54
            	Error:      	Received unexpected error:
            	            	listen udp 127.0.0.1:50408: bind: An attempt was made to access a socket in a way forbidden by its access permissions.
            	Test:       	Test_Server_ListenAndServe/udp
FAIL
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/statsdreceiver/transport	0.045s
```

--- a/receiver/statsdreceiver/transport/server_test.go
+++ b/receiver/statsdreceiver/transport/server_test.go
@@ -31,6 +31,8 @@ import (
 )
 
 func Test_Server_ListenAndServe(t *testing.T) {
+	t.Skip("Test is unstable, see https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1426")
+
 	tests := []struct {
 		name          string
 		buildServerFn func(addr string) (Server, error)

